core/test/variable_operations_test.cpp

```diff
@@ -143,6 +143,19 @@ TEST(Variable, operator_plus_eigen_type) {
   EXPECT_EQ(sum.dtype(), dtype<Eigen::Vector3d>);
 }
 
+TEST(SparseVariable, operator_plus) {
```

Comment:
Is there any reason to prefer the non-tempated operator approach (here) to the one now used in `dataset_operatorations_test`?

Reply:
Yes, we have nothing to compare to, so we cannot construct outputs readily without hand-writing them. But as mentioned in the PR message I am looking into cleaning this up more as part of improving the tests for `transform`.